llvm/test/CodeGen/EraVM/array.ll

```diff
@@ -19,15 +19,15 @@ define i256 @consti_loadconst_neg_offset(i256 %idx) nounwind {
 
 ; CHECK-LABEL: consti_loadconst_storeglobal
 define void @consti_loadconst_storeglobal() nounwind {
-  ; CHECK: add @const[1], r0, stack[@val+1]
+  ; CHECK: add @const[r0+1], r0, stack[@val + 1]
```

Comment:
Is this desired? Can we skip `r0` here for the sake of clarity?

Reply:
The idea is to simplify asm printing first (without changing too many tests, so reviewing test changes is manageable). Then, it should be easier to improve the formatting via mostly obvious changes in `EraVMInstPrinter.cpp` (but there will be lots of changes in tests).

This PR strives to collapse the original operand printing code into a call to `analyzeMCOperandsCode` or `analyzeMCOperandsStack` and a _mostly_ straightforward formatting.

Printing `r0` here is not desired, of course. At first glance it looked like a good balance between temporarily overcomplicating printer and changing many existing tests. Now I experimented a bit more and it seems things can be simplified a bit more - I will update this PR tomorrow.

Reply:
Fixed